### PR TITLE
XW-3854 | Make sure videos only autoplay on active tabs

### PR DESCRIPTION
**Pull request by JamesHawking**
[Ticket](https://wikia-inc.atlassian.net/browse/XW-3854)

### CODE REVIEW

File: extensions/wikia/ArticleVideo/scripts/article-featured-video.js [hunk 1]
```diff
@@ -55,7 +59,10 @@ require([
 			videoLabels = (videoData.labels || '').join(','),
 			videoFeedbackBox,
 			autoplayCookieName = 'featuredVideoAutoplay',
-			autoplay = cookies.get(autoplayCookieName) !== '0';
+			autoplayEnabled = cookies.get(autoplayCookieName) !== '0' &&
+				geo.isProperGeo(instantGlobals.wgArticleVideoAutoplayCountries);
+			autoplay = autoplayEnabled && !document.hidden,
```
Comment (WikiaTech) [inline extensions/wikia/ArticleVideo/scripts/article-featured-video.js]: MAJOR Add the "let", "const" or "var" keyword to this declaration of "autoplay" to make it explicit. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2703)

File: extensions/wikia/ArticleVideo/scripts/article-featured-video.js [hunk 2]
```diff
@@ -55,7 +59,10 @@ require([
...
+			playerIsAfterInitialPlay = false;
```
Comment (WikiaTech) [inline extensions/wikia/ArticleVideo/scripts/article-featured-video.js]: MAJOR Add the "let", "const" or "var" keyword to this declaration of "playerIsAfterInitialPlay" to make it explicit. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2703)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/ArticleVideo/scripts/article-featured-video.js [hunk 3]
```diff
@@ -59,8 +59,10 @@ require([
 			videoLabels = (videoData.labels || '').join(','),
 			videoFeedbackBox,
 			autoplayCookieName = 'featuredVideoAutoplay',
-			autoplay = cookies.get(autoplayCookieName) !== '0' &&
+			autoplayEnabled = cookies.get(autoplayCookieName) !== '0' &&
 				geo.isProperGeo(instantGlobals.wgArticleVideoAutoplayCountries);
+			autoplay = autoplayEnabled && !document.hidden,
```
Comment (WikiaTech) [inline extensions/wikia/ArticleVideo/scripts/article-featured-video.js]: MAJOR Add the "let", "const" or "var" keyword to this declaration of "autoplay" to make it explicit. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2703)

File: extensions/wikia/ArticleVideo/scripts/article-featured-video.js [hunk 4]
```diff
@@ -59,8 +59,10 @@ require([
...
+			playerIsAfterInitialPlay = false;
```
Comment (WikiaTech) [inline extensions/wikia/ArticleVideo/scripts/article-featured-video.js]: MAJOR Add the "let", "const" or "var" keyword to this declaration of "playerIsAfterInitialPlay" to make it explicit. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2703)

**Review by latata — APPROVED** [review]

File: extensions/wikia/ArticleVideo/scripts/article-featured-video.js [hunk 5]
```diff
@@ -59,8 +59,10 @@ require([
 			videoLabels = (videoData.labels || '').join(','),
 			videoFeedbackBox,
 			autoplayCookieName = 'featuredVideoAutoplay',
-			autoplay = cookies.get(autoplayCookieName) !== '0' &&
-				geo.isProperGeo(instantGlobals.wgArticleVideoAutoplayCountries);
+			autoplayEnabled = cookies.get(autoplayCookieName) !== '0' &&
+				geo.isProperGeo(instantGlobals.wgArticleVideoAutoplayCountries),
+			autoplay = autoplayEnabled && !document.hidden,
+			playerIsAfterInitialPlay = false;
```
Comment (rogatty) [inline extensions/wikia/ArticleVideo/scripts/article-featured-video.js]: How about `initialPlayTriggered`? I think it's more descriptive.

File: extensions/wikia/ArticleVideo/scripts/article-featured-video.js [hunk 6]
```diff
@@ -226,6 +236,8 @@ require([
 			}
 
 			player.mb.subscribe(window.OO.EVENTS.INITIAL_PLAY, 'featured-video', function () {
+				initialPlayTriggered = true;
```
Comment (WikiaTech) [inline extensions/wikia/ArticleVideo/scripts/article-featured-video.js]: MAJOR Remove this useless assignment to local variable "initialPlayTriggered" [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1854)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/ArticleVideo/scripts/article-featured-video.js [hunk 7]
```diff
@@ -61,7 +61,9 @@ require([
 			inAutoplayCountries = geo.isProperGeo(instantGlobals.wgArticleVideoAutoplayCountries),
 			inNextVideoAutoplayCountries = geo.isProperGeo(instantGlobals.wgArticleVideoNextVideoAutoplayCountries),
 			autoplayCookieName = 'featuredVideoAutoplay',
-			autoplay = cookies.get(autoplayCookieName) !== '0' && inAutoplayCountries;
+			autoplayEnabled = cookies.get(autoplayCookieName) !== '0' && inAutoplayCountries,
+			autoplay = autoplayEnabled && !document.hidden,
```
Comment (rogatty) [inline extensions/wikia/ArticleVideo/scripts/article-featured-video.js]: We do `window.guaSetCustomDimension(37, autoplay ? 'Yes' : 'No');` so this change would pollute the stats.

**Review by rogatty — APPROVED** [review]

**Pull request merged**